src/java/org/apache/cassandra/tcm/ClusterMetadata.java

```diff
@@ -817,6 +818,15 @@ public static ClusterMetadata currentNullable()
         return service.metadata();
     }
 
+    public static Optional<ClusterMetadata> currentOptional()
```

Comment:
should we migrate any `currentNullable()` uses to this instead? Or just use `currentNullable` in `scanLogForPeriod`? (which will soon be gone anyway)